src/dev/flang/be/c/C.java

```diff
@@ -164,11 +164,12 @@ public CStmnt assign(int cl, int c, int i, CExpr tvalue, CExpr avalue)
      */
     public Pair<CExpr, CStmnt> call(int cl, int c, int i, CExpr tvalue, List<CExpr> args)
     {
-      var cc0 = _fuir.accessedClazz  (cl, c, i);
+      var ccP = _fuir.accessedPreconditionClazz(cl, c, i);
+      var cc0 = _fuir.accessedClazz            (cl, c, i);
       var ol = new List<CStmnt>();
-      if (_fuir.hasPrecondition(cc0))
+      if (ccP != -1)
```

Comment:
_Magic_ number should be put in a constant imho.

Reply:
-1 is used at several places to mean 'no class', could be fixed in a separate patch.